src/directives/justgage.js

```diff
@@ -85,6 +102,14 @@ angular.module('frapontillo.gage.directives', ['frapontillo.gage.controllers'])
                 justgage.refresh(scope.value, newValue);
               }
             }));
+            // will fire when the justgage element is brought back into view
+            // if something slides it off the screen or sets ng-show/hide this will ensure the justgage is redrawn and animated properly
+            watchers.push(scope.$watch(function () { return isElementInViewPort(element[0]); }, function (newValue) {
```

Comment:
Updated - new attribute, watch-view-port, set to true to set up that watcher, false to not.  It wouldn't response to changes, not 100% sure how I'd make that work